### PR TITLE
ticdc: add sync point config

### PR DESCRIPTION
**Pull request by asddongmen**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v6.3 (TiDB 6.3 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: ticdc/manage-ticdc.md [hunk 1]
```diff
@@ -628,6 +628,13 @@ case-sensitive = true
 # 是否输出 old value，从 v4.0.5 开始支持，从 v5.0 开始默认为 true
 enable-old-value = true
 
+# 是否开启 sync point 功能，从 v6.3.0 开始支持
+enable-sync-point = true
+# sync point 功能对齐上下游 snapshot 的时间间隔，最小值为 30s
```
Comment (nongfushanquan) [inline ticdc/manage-ticdc.md:633]: add comments to enum the format of the values( sync-point-interval,sync-point- retention)

File: sync-diff-inspector/upstream-downstream-diff.md [hunk 2]
```diff
@@ -38,18 +45,18 @@ select * from tidb_binlog.checkpoint;
...
     user = "root"
     password = ""
-    snapshot = "409621863377928345"
+    snapshot = "435953235516456963"
 ```
 
 ## 注意事项
 
-- Drainer 的 `db-type` 需要设置为 `tidb`，这样才会在 checkpoint 中保存 `ts-map`。
+- TiCDC 在创建 Changefeed 时， `enable-sync-point` 需要设置为 `true`，这样才会开启 sync point 功能，在下游保存 `ts-map`。完整的配置请参考 [TiCDC 同步任务配置文件描述](/ticdc/manage-ticdc.md#同步任务配置文件描述)
```
Comment (TomShawn) [inline sync-diff-inspector/upstream-downstream-diff.md:60]:
```suggestion
- TiCDC 在创建 Changefeed 前，请确保 TiCDC 的配置项 `enable-sync-point` 已设置为 `true`，这样才会开启 sync point 功能，在下游保存 `ts-map`。完整的配置请参考 [TiCDC 同步任务配置文件描述](/ticdc/manage-ticdc.md#同步任务配置文件描述)。
```

File: ticdc/manage-ticdc.md [hunk 3]
```diff
@@ -628,6 +628,17 @@ case-sensitive = true
...
+# 是否开启 sync point 功能，从 v6.3.0 开始支持
+enable-sync-point = true
+# sync point 功能对齐上下游 snapshot 的时间间隔
+# 配置格式为 h m s, 例如 "1h30m30s"
+# 默认值为 10m，最小值为 30s
+sync-point-interval = "5m"
+# sync point 功能在下游表中保存的数据的时长，超过这个时间的数据会被清理
+# 配置格式为 h m s, 例如 "24h30m30s"
+# 默认值为 24h
+sync-point-retention = "1h"
```
Comment (TomShawn) [inline ticdc/manage-ticdc.md:640]:
```suggestion
# 是否开启 sync point 功能，从 v6.3.0 开始支持
enable-sync-point = true

# sync point 功能对齐上下游 snapshot 的时间间隔
# 配置格式为 h m s，例如 "1h30m30s"
# 默认值为 10m，最小值为 30s
sync-point-interval = "5m"

# sync point 功能在下游表中保存的数据的时长，超过这个时间的数据会被清理
# 配置格式为 h m s，例如 "24h30m30s"
# 默认值为 24h
sync-point-retention = "1h"
```

**Comment by TomShawn** [comment]
感觉需要在文档里单独介绍下什么是 sync point，使用场景，以及简单提一下原理？

File: sync-diff-inspector/upstream-downstream-diff.md [hunk 4]
```diff
@@ -5,7 +5,7 @@ aliases: ['/docs-cn/dev/sync-diff-inspector/upstream-downstream-diff/','/docs-cn
 
 # TiDB 主从集群的数据校验
 
-在你使用 TiDB Binlog 搭建 TiDB 的主从集群，Drainer 在把数据同步到 TiDB 时，保存 checkpoint 的同时也会将上下游的 TSO 对应关系保存为 `ts-map`。在 sync-diff-inspector 中配置 `snapshot` 即可对 TiDB 主从集群的数据进行校验。
+在你使用 TiCDC 搭建 TiDB 的主从集群时，如果开启了 sync point 功能，那么 TiCDC 在数据的同步过程中会根据你所配置的 `sync-point-interval` 来定时对齐上下游的 snapshot, 并将上下游的 TSO 对应关系保存为 `ts-map`。在 sync-diff-inspector 中配置 `snapshot` 即可对 TiDB 主从集群的数据进行校验。
```
[the remark this answers is not in the document]
Comment (leoppro) [reply sync-diff-inspector/upstream-downstream-diff.md:8]: For @asddongmen :
- 可以在第一段一两句话介绍sync-point 功能然后引用sync-point操作文档。
- 建议写一下如何创建一个带sync-point-interval 配置的changefeed
For @TomShawn 
- Binlog 已经不维护，因此产研在逐步使用 TiCDC 替代 Binlog

[the remark this answers is not in the document]
Comment (asddongmen) [reply sync-diff-inspector/upstream-downstream-diff.md:8]: ok

File: sync-diff-inspector/upstream-downstream-diff.md [hunk 5]
```diff
@@ -5,7 +5,22 @@ aliases: ['/docs-cn/dev/sync-diff-inspector/upstream-downstream-diff/','/docs-cn
 
 # TiDB 主从集群的数据校验
 
-在你使用 TiDB Binlog 搭建 TiDB 的主从集群，Drainer 在把数据同步到 TiDB 时，保存 checkpoint 的同时也会将上下游的 TSO 对应关系保存为 `ts-map`。在 sync-diff-inspector 中配置 `snapshot` 即可对 TiDB 主从集群的数据进行校验。
+在正常的同步模式中，TiCDC 只提供了数据的最终一致性的保证，而无法确保在同步的过程中数据的一致性。当你使用 TiCDC 搭建 TiDB 的主从集群时，可能会需要在不停止同步的情况下中对上下游的数据进行一致性验证。然而，对动态变更的数据进行一致性校验是非常困难的，为了满足这一需求，TiCDC 提供了 Syncpoint 功能。
```
Comment (TomShawn) [inline sync-diff-inspector/upstream-downstream-diff.md:8]:
```suggestion
当你使用 TiCDC 搭建 TiDB 的主从集群时，可能会需要在不停止同步的情况下对上下游的数据进行一致性验证。在普通的同步模式中，TiCDC 只提供数据的最终一致性的保证，而无法确保在同步的过程中数据的一致性。因此，对动态变更的数据进行一致性校验非常困难，为了满足这一需求，TiCDC 提供了 Syncpoint 功能。
```

File: sync-diff-inspector/upstream-downstream-diff.md [hunk 6]
```diff
@@ -5,7 +5,22 @@ aliases: ['/docs-cn/dev/sync-diff-inspector/upstream-downstream-diff/','/docs-cn
...
+
+Syncpoint 通过利用 TiDB 提供的 snapshot 特性，让 TiCDC 在同步过程维护了一个上下游具有一致性 snapshot 的 `ts-map`。把动态的数据一致性校验问题转化成了静态的 snapshot 数据一致性校验问题，达到了接近数据一致性实时校验的效果。
```
Comment (TomShawn) [inline sync-diff-inspector/upstream-downstream-diff.md:10]:
```suggestion
Syncpoint 通过利用 TiDB 提供的 snapshot 特性，让 TiCDC 在同步过程中维护了一个上下游具有一致性 snapshot 的 `ts-map`。把校验动态数据的一致性问题转化为了校验静态 snapshot 数据的一致性问题，达到了接近数据一致性实时校验的效果。
```

File: sync-diff-inspector/upstream-downstream-diff.md [hunk 7]
```diff
@@ -5,7 +5,22 @@ aliases: ['/docs-cn/dev/sync-diff-inspector/upstream-downstream-diff/','/docs-cn
...
+
+你可以在创建同步任务时把 `enable-sync-point` 设置为 `true` 来开启 SyncPoint 功能。如果开启了 Syncpoint 功能，那么 TiCDC 在数据的同步过程中会根据你所配置的 `sync-point-interval` 来定时对齐上下游的 snapshot, 并将上下游的 TSO 对应关系保存在下游的 `tidb_cdc.syncpoint_v1` 表中。
```
Comment (TomShawn) [inline sync-diff-inspector/upstream-downstream-diff.md:12]:
```suggestion
要开启 SyncPoint 功能，你可以在创建同步任务时把 TiCDC 的配置项 `enable-sync-point` 设置为 `true`。开启 Syncpoint 功能后，TiCDC 在数据的同步过程中会根据你所配置的 TiCDC 参数 `sync-point-interval` 来定时对齐上下游的 snapshot，并将上下游的 TSO 对应关系保存在下游的 `tidb_cdc.syncpoint_v1` 表中。
```

File: sync-diff-inspector/upstream-downstream-diff.md [hunk 8]
```diff
@@ -5,7 +5,22 @@ aliases: ['/docs-cn/dev/sync-diff-inspector/upstream-downstream-diff/','/docs-cn
...
+
+然后，你只需要在 sync-diff-inspector 中配置 `snapshot` 即可对 TiDB 主从集群的数据进行校验。以下为创建一个开启 Syncpoint 功能的同步任务的配置示例。
```
Comment (TomShawn) [inline sync-diff-inspector/upstream-downstream-diff.md:14]:
```suggestion
然后，你只需要在 sync-diff-inspector 中配置 `snapshot` 即可对 TiDB 主从集群的数据进行校验。以下配置示例为创建的同步任务开启 Syncpoint 功能：
```

File: sync-diff-inspector/upstream-downstream-diff.md [hunk 9]
```diff
@@ -5,7 +5,22 @@ aliases: ['/docs-cn/dev/sync-diff-inspector/upstream-downstream-diff/','/docs-cn
...
+
+```toml
+# 开启 SyncPoint
+enable-sync-point = true
+# 每隔 5 分钟对齐一次上下游的 snapshot
+sync-point-interval = "5m"
+# 每隔 1 小时清理一次下游 tidb_cdc.syncpoint_v1 表中的 ts-map 数据
+sync-point-retention = "1h"
```
Comment (TomShawn) [inline sync-diff-inspector/upstream-downstream-diff.md:22]:
```suggestion
# 开启 SyncPoint
enable-sync-point = true

# 每隔 5 分钟对齐一次上下游的 snapshot
sync-point-interval = "5m"

# 每隔 1 小时清理一次下游 tidb_cdc.syncpoint_v1 表中的 ts-map 数据
sync-point-retention = "1h"
```

File: sync-diff-inspector/upstream-downstream-diff.md [hunk 10]
```diff
@@ -14,18 +29,25 @@ aliases: ['/docs-cn/dev/sync-diff-inspector/upstream-downstream-diff/','/docs-cn
...
 ```
 
-从结果中可以获取 ts-map 信息。
+以上 `syncpoint_v1` 表中各列所代表的信息如下:
+
+- `ticdc_cluster_id`: 插入该条记录的 TiCDC 集群的 ID。
+- `changefeed`: 插入该条记录的 Changefeed 的 ID，由于不同的 TiCDC 集群可能会存在重名的 Changefeed，所以需要通过 TiCDC 集群 ID 和 Changefeed 的 ID 来唯一确认一个 Changefeed 所插入的 `ts-map`。
+- `primary_ts`: 上游数据库 snapshot 的 ts。
+- `secondary_ts`: 下游数据库 snapshot 的 ts。
+- `created_at`: 插入该条记录的时间。
```
Comment (TomShawn) [inline sync-diff-inspector/upstream-downstream-diff.md:49]:
```suggestion
- `ticdc_cluster_id`：插入该条记录的 TiCDC 集群的 ID。
- `changefeed`：插入该条记录的 Changefeed 的 ID。由于不同的 TiCDC 集群可能会存在重名的 Changefeed，所以需要通过 TiCDC 集群 ID 和 Changefeed 的 ID 来确认一个 Changefeed 所插入的唯一 `ts-map`。
- `primary_ts`：上游数据库 snapshot 的时间戳。
- `secondary_ts`：下游数据库 snapshot 的时间戳。
- `created_at`：插入该条记录的时间。
```

File: sync-diff-inspector/upstream-downstream-diff.md [hunk 11]
```diff
@@ -38,18 +60,18 @@ select * from tidb_binlog.checkpoint;
...
     user = "root"
     password = ""
-    snapshot = "409621863377928345"
+    snapshot = "435953235516456963"
 ```
 
 ## 注意事项
 
-- Drainer 的 `db-type` 需要设置为 `tidb`，这样才会在 checkpoint 中保存 `ts-map`。
+- TiCDC 在创建 Changefeed 前，请确保 TiCDC 的配置项 `enable-sync-point` 已设置为 `true`，这样才会开启 sync point 功能，在下游保存 `ts-map`。完整的配置请参考 [TiCDC 同步任务配置文件描述](/ticdc/manage-ticdc.md#同步任务配置文件描述)。
```
Comment (TomShawn) [inline sync-diff-inspector/upstream-downstream-diff.md:75]:
```suggestion
- TiCDC 在创建 Changefeed 前，请确保 TiCDC 的配置项 `enable-sync-point` 已设置为 `true`，这样才会开启 Syncpoint 功能，在下游保存 `ts-map`。完整的配置请参考 [TiCDC 同步任务配置文件描述](/ticdc/manage-ticdc.md#同步任务配置文件描述)。
```

File: ticdc/manage-ticdc.md [hunk 12]
```diff
@@ -628,6 +628,19 @@ case-sensitive = true
...
+enable-sync-point = true
+
+# sync point 功能对齐上下游 snapshot 的时间间隔
+# 配置格式为 h m s，例如 "1h30m30s"
+# 默认值为 10m，最小值为 30s
+sync-point-interval = "5m"
+
+# sync point 功能在下游表中保存的数据的时长，超过这个时间的数据会被清理
+# 配置格式为 h m s，例如 "24h30m30s"
+# 默认值为 24h
```
Comment (TomShawn) [inline ticdc/manage-ticdc.md:641]:
```suggestion
# 是否开启 Syncpoint 功能，从 v6.3.0 开始支持
enable-sync-point = true

# Syncpoint 功能对齐上下游 snapshot 的时间间隔
# 配置格式为 h m s，例如 "1h30m30s"
# 默认值为 10m，最小值为 30s
sync-point-interval = "5m"

# Syncpoint 功能在下游表中保存的数据的时长，超过这个时间的数据会被清理
# 配置格式为 h m s，例如 "24h30m30s"
# 默认值为 24h
```

**Review by TomShawn — APPROVED** [review]
LGTM

File: sync-diff-inspector/upstream-downstream-diff.md [hunk 13]
```diff
@@ -14,18 +31,25 @@ aliases: ['/docs-cn/dev/sync-diff-inspector/upstream-downstream-diff/','/docs-cn
...
++------------------+----------------+--------------------+--------------------+---------------------+
+| default          | default_test-2 | 435953225454059520 | 435953235516456963 | 2022-09-13 08:40:15 |
++------------------+----------------+--------------------+--------------------+---------------------+
 ```
 
-从结果中可以获取 ts-map 信息。
+以上 `syncpoint_v1` 表中各列所代表的信息如下:
+
+- `ticdc_cluster_id`：插入该条记录的 TiCDC 集群的 ID。
+- `changefeed`：插入该条记录的 Changefeed 的 ID。由于不同的 TiCDC 集群可能会存在重名的 Changefeed，所以需要通过 TiCDC 集群 ID 和 Changefeed 的 ID 来确认一个 Changefeed 所插入的唯一 `ts-map`。
```
Comment (TomShawn) [inline sync-diff-inspector/upstream-downstream-diff.md:48]:
```suggestion
- `changefeed`：插入该条记录的 Changefeed 的 ID。由于不同的 TiCDC 集群可能会存在重名的 Changefeed，所以需要通过 TiCDC 集群 ID 和 Changefeed 的 ID 来确认一个 Changefeed 所插入的`ts-map`。
```

File: sync-diff-inspector/upstream-downstream-diff.md [hunk 14]
```diff
@@ -14,18 +31,25 @@ aliases: ['/docs-cn/dev/sync-diff-inspector/upstream-downstream-diff/','/docs-cn
...
++------------------+----------------+--------------------+--------------------+---------------------+
+| default          | default_test-2 | 435953225454059520 | 435953235516456963 | 2022-09-13 08:40:15 |
++------------------+----------------+--------------------+--------------------+---------------------+
 ```
 
-从结果中可以获取 ts-map 信息。
+以上 `syncpoint_v1` 表中各列所代表的信息如下:
+
+- `ticdc_cluster_id`：插入该条记录的 TiCDC 集群的 ID。
+- `changefeed`：插入该条记录的 Changefeed 的 ID。由于不同的 TiCDC 集群可能会存在重名的 Changefeed，所以需要通过 TiCDC 集群 ID 和 Changefeed 的 ID 来确认一个 Changefeed 所插入的`ts-map`。
```
Comment (TomShawn) [inline sync-diff-inspector/upstream-downstream-diff.md:48]:
```suggestion
- `changefeed`：插入该条记录的 Changefeed 的 ID。由于不同的 TiCDC 集群可能会存在重名的 Changefeed，所以需要通过 TiCDC 集群 ID 和 Changefeed 的 ID 来确认一个 Changefeed 所插入的 `ts-map`。
```

File: sync-diff-inspector/upstream-downstream-diff.md [hunk 15]
```diff
@@ -14,18 +31,25 @@ aliases: ['/docs-cn/dev/sync-diff-inspector/upstream-downstream-diff/','/docs-cn
 {{< copyable "sql" >}}
 
 ```sql
-select * from tidb_binlog.checkpoint;
+select * from tidb_cdc.syncpoint_v1;;
```
Comment (TomShawn) [inline sync-diff-inspector/upstream-downstream-diff.md:34]:
```suggestion
select * from tidb_cdc.syncpoint_v1;
```

File: sync-diff-inspector/upstream-downstream-diff.md [hunk 16]
```diff
@@ -14,18 +31,25 @@ aliases: ['/docs-cn/dev/sync-diff-inspector/upstream-downstream-diff/','/docs-cn
...
 
-从结果中可以获取 ts-map 信息。
+以上 `syncpoint_v1` 表中各列所代表的信息如下:
+
+- `ticdc_cluster_id`：插入该条记录的 TiCDC 集群的 ID。
+- `changefeed`：插入该条记录的 Changefeed 的 ID。由于不同的 TiCDC 集群可能会存在重名的 Changefeed，所以需要通过 TiCDC 集群 ID 和 Changefeed 的 ID 来确认一个 Changefeed 所插入的 `ts-map`。
+- `primary_ts`：上游数据库 snapshot 的时间戳。
+- `secondary_ts`：下游数据库 snapshot 的时间戳。
+- `created_at`：插入该条记录的时间。
+
```
Comment (TomShawn) [inline sync-diff-inspector/upstream-downstream-diff.md:52]:
```suggestion
```

File: sync-diff-inspector/upstream-downstream-diff.md [hunk 17]
```diff
@@ -5,7 +5,24 @@ aliases: ['/docs-cn/dev/sync-diff-inspector/upstream-downstream-diff/','/docs-cn
...
 # TiDB 主从集群的数据校验
 
-在你使用 TiDB Binlog 搭建 TiDB 的主从集群，Drainer 在把数据同步到 TiDB 时，保存 checkpoint 的同时也会将上下游的 TSO 对应关系保存为 `ts-map`。在 sync-diff-inspector 中配置 `snapshot` 即可对 TiDB 主从集群的数据进行校验。
+当你使用 TiCDC 搭建 TiDB 的主从集群时，可能会需要在不停止同步的情况下对上下游的数据进行一致性验证。在普通的同步模式中，TiCDC 只提供数据的最终一致性的保证，而无法确保在同步的过程中数据的一致性。因此，对动态变更的数据进行一致性校验非常困难，为了满足这一需求，TiCDC 提供了 Syncpoint 功能。
+
+Syncpoint 通过利用 TiDB 提供的 snapshot 特性，让 TiCDC 在同步过程中维护了一个上下游具有一致性 snapshot 的 `ts-map`。把校验动态数据的一致性问题转化为了校验静态 snapshot 数据的一致性问题，达到了接近数据一致性实时校验的效果。
+
+要开启 SyncPoint 功能，你可以在创建同步任务时把 TiCDC 的配置项 `enable-sync-point` 设置为 `true`。开启 Syncpoint 功能后，TiCDC 在数据的同步过程中会根据你所配置的 TiCDC 参数 `sync-point-interval` 来定时对齐上下游的 snapshot，并将上下游的 TSO 对应关系保存在下游的 `tidb_cdc.syncpoint_v1` 表中。
+
+然后，你只需要在 sync-diff-inspector 中配置 `snapshot` 即可对 TiDB 主从集群的数据进行校验。以下 TiCDC 配置示例为创建的同步任务开启 Syncpoint 功能：
```
Comment (shichun-0415) [inline sync-diff-inspector/upstream-downstream-diff.md:14]: 可以去掉”主从“或者把它改为 "上下游"么？

**Review by shichun-0415 — APPROVED** [review]
Rest LGTM

File: sync-diff-inspector/upstream-downstream-diff.md [hunk 18]
```diff
@@ -5,7 +5,24 @@ aliases: ['/docs-cn/dev/sync-diff-inspector/upstream-downstream-diff/','/docs-cn
 
 # TiDB 主从集群的数据校验
 
-在你使用 TiDB Binlog 搭建 TiDB 的主从集群，Drainer 在把数据同步到 TiDB 时，保存 checkpoint 的同时也会将上下游的 TSO 对应关系保存为 `ts-map`。在 sync-diff-inspector 中配置 `snapshot` 即可对 TiDB 主从集群的数据进行校验。
+当你使用 TiCDC 搭建 TiDB 的主从集群时，可能会需要在不停止同步的情况下对上下游的数据进行一致性验证。在普通的同步模式中，TiCDC 只提供数据的最终一致性的保证，而无法确保在同步的过程中数据的一致性。因此，对动态变更的数据进行一致性校验非常困难，为了满足这一需求，TiCDC 提供了 Syncpoint 功能。
```
[the remark this answers is not in the document]
Comment (TomShawn) [reply sync-diff-inspector/upstream-downstream-diff.md:8]: They are not the same pair of concepts.

**Pull request merged**